hypha/static_src/src/app/src/containers/MetaTerm/containers/MetaTermForm/saga.js

```diff
@@ -0,0 +1,57 @@
+import {
+    call,
+    put,
+    takeLatest,
+  } from 'redux-saga/effects';
+  import * as ActionTypes from './constants';
+  import * as Actions from './actions';
+  import { updateSubmissionMetaTerms } from '@actions/submissions'
+  import { apiFetch } from '@api/utils';
+  import { camelizeKeys } from 'humps';
+
+  
+  export function* initialFetch() {
+    
+    try {
+      yield put(Actions.showLoadingAction())
+      let response = yield call(apiFetch, {path : `/v1/meta_terms/`});
+      let data = yield response.json()
+      yield put(
+        Actions.getMetaTermsSuccessAction(data.results),
+      );
+      yield put(Actions.hideLoadingAction())
+  
+    } catch (e) {
+      yield put(Actions.hideLoadingAction())
+    }
+  }
+  
+  export function* updateMetaTerms(action){
+    const url = `/v1/submissions/${action.submissionId}/meta_terms/`
+    try{
+      yield put(Actions.showLoadingAction())
+      yield call(
+        apiFetch, 
+        {
+          path : url,
+          method : "POST",
+          options : {
+            body : JSON.stringify({"meta_terms" : action.data}),
+            }
+          }
+        )
+      let response = yield call(apiFetch, {path : `/v1/submissions/${action.submissionId}`});
+      response = yield response.json()
+      const data = yield camelizeKeys(response)
+      yield put(updateSubmissionMetaTerms(action.submissionId, data.metaTerms))
```

Comment:
Wouldn't the submission already have metaTerms? Looking at `SubmissionDetailSerializer`.

Reply:
I make sure when user updates meta term, redux store should be updated in sync with database. Sometimes, update to database(POST) may fail, which causes different data in the store & database. So, updated meta term after fetching data from detail API.

Reply:
Not required, If you see line no 39, I'm updating the meta term alone.